invokeai/app/invocations/image.py

```diff
@@ -118,6 +124,80 @@ def invoke(self, context: InvocationContext) -> ImageOutput:
             image=image,
         )
 
+class DownloadImageInvocation(BaseInvocation):
+    """Fetches an image from a URL and provides it as output."""
+
+    type: Literal["download_image"] = "download_image"
+
+    # Inputs
+    image_url: str = Field(description="The URL of the image to download")
+
+    def invoke(self, context: InvocationContext) -> ImageOutput:
+        r = requests.get(self.image_url)
+        a = urlparse(self.image_url)
+
+        # TODO: how to handle failure?
+        if r.ok and r.headers["content-type"].startswith("image/"):
+            raw_image = io.BytesIO()
+            for chunk in r.iter_content(chunk_size=128):
+                raw_image.write(chunk)
+            raw_image.flush()
+            raw_image.seek(0)
+
+            return build_image_output(
+                image_type=ImageType.UPLOAD,
+                image_name=os.path.basename(a.path),
+                image=Image.Image.frombytes(raw_image)
+            )
+
+RESIZE_RESAMPLING_MODE = Literal[
+    "nearest", "box", "bilinear", "hamming", "bicubic", "lanczos"
+]
+
+class ResizeImageInvocation(BaseInvocation, PILInvocationConfig):
+    """Resizes an image to a specified size."""
+
+    type: Literal["resize"] = "resize"
+
+    # Inputs
+    image: ImageField = Field(default=None, description="The image to resize")
+    size: int = Field(default=512, gt=0, description="The size of the resized image's longest side")
```

Comment:
Is there a reason for not accepting a width and height? What if I want to stretch the image to a new aspect ratio?

Reply:
We could do that. My aim was to be able to scale an image down to ensure it won't overwhelm VRAM when fed into img2img